src/gen/gen.c

```diff
@@ -232,6 +233,26 @@ static void yf_gen_if(
 
 int yfg_gen(struct yf_compile_analyse_job * data, struct yf_gen_info * info) {
 
+    /**
+     * Because we create the output file in the form of 'bin/c/foo/bar/baz.yf',
+     * we need to first mkdir() the enclosing directory, if it doesn't already
+     * exist. To do that, we would replace the last slash with a \0, mkdir() the
+     * path, and set it back.
+     * However, we have to create ALL enclosing folders first, so we need to get
+     * all slashes from left to right, and do this recursively.
+     */
```

Comment:
I think this could be moved to a separate function.

Reply:
Definitely agree.